### DISCUSSION TITLE
Use Groovy 4.0

### DISCUSSION THREAD

**Discussion by fletchgqc** [qualersoft/jmeter-gradle-plugin#213] [Q&A] [answered]
I needed Groovy 4 for my project. When using jmeter standalone, I just substituted the groovy-3 jars in the libs folder for the equivalent groovy-4 jars and everything worked. How could I achieve this with jmeter-gradle-plugin?

**Comment by mathze** [comment, marked as answer]
Hi @fletchgqc,

puh this was a hard on, normaly you simply can use gradle build in functionality like described at https://docs.gradle.org/current/userguide/dependency_constraints.html

The issue with groovy is, they changed the group-name from `org.codehouse.groovy` to `org.apache.groovy` so normal gradle dependency-overwriting does not work:/

But it's gradle, so there is always a way^^
long story short...
Put this snippet in your build.gradle.kts (attention I used kotlin-dsl, but should not mutch different in groovy-dsl)
```
configurations.jmeterPlugin {
  resolutionStrategy.eachDependency {
    if (requested.group == "org.codehaus.groovy") {
      useTarget("org.apache.groovy:${requested.name}:4.0.0")
    }
  }
}
```
Maybe gradle offers a more elegant way, if this method cause issues you should have a look at https://docs.gradle.org/current/userguide/resolution_rules.html

Last but not least, a word of warning from personal experience: Tampering with groovy-versions may raise nondeterministic issues especially when it came to type-resolution.
So it might be better to wait until jmeter upgrades to groovy 4.

I hope this helps and no issues🤞

**Reply by fletchgqc** [reply]
Wow, thanks a lot Mathze!
